src/qutip_tensornetwork/core/data/network.py

```diff
@@ -576,6 +576,88 @@ def tensor(self, other):
 
         return Network._fast_constructor(out_edges, in_edges, nodes)
 
+    def match_out_dims(self, target_dims):
+        """Reshape nodes by splitting edges such that `in_edges` matches
+        ``target_dims``. After this function the following will hold:
+            ``network.dims[0] == target_dims``
+
+        Note
+        ----
+        This is an in-place operation that returns itself.
```

Comment:
I do not have any strong reason to make it return self. I do like the fact that the following code:
```python
network.match_out_dims() 
```
will print the network. I can remove it if you believe it can cause any problems.

Reply:
I don't think it will directly cause problems, I just think it will surprise Python people a bit that `network.match_out/in_dims(...)` will both modify `network` and return it. :)